auth/src/main/java/com/firebase/ui/auth/ui/email/WelcomeBackPasswordPrompt.java

```diff
@@ -121,45 +126,44 @@ public void onChanged(@Nullable PendingResolution resolution) {
                 });
 
         // Observe the state of the main auth operation
-        mHandler.getSignInResult().observe(this, new Observer<Resource<IdpResponse>>() {
+        mHandler.getSignInOperation().observe(this, new Observer<Resource<IdpResponse>>() {
             @Override
             public void onChanged(@Nullable Resource<IdpResponse> resource) {
-                onAuthResult(resource);
+                onSignInOperation(resource);
             }
         });
     }
 
     @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        // Forward activity results to the ViewModel
-        if (!mHandler.onActivityResult(requestCode, resultCode, data)) {
-            super.onActivityResult(requestCode, resultCode, data);
-        }
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putString(ExtraConstants.EXTRA_PASSWORD, mPendingPassword);
     }
 
-    private void onAuthResult(@Nullable Resource<IdpResponse> resource) {
+    private void onSignInOperation(@Nullable Resource<IdpResponse> resource) {
         if (resource == null) {
             Log.w(TAG, "Got null resource, ignoring.");
             return;
         }
 
-        if (resource.getState() == State.LOADING) {
-            getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_in);
-        }
-
-        if (resource.getState() == State.SUCCESS) {
-            getDialogHolder().dismissDialog();
-            Log.d(TAG, "onAuthResult:SUCCESS:" + resource.getValue());
+        switch (resource.getState()) {
+            case LOADING:
+                getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_in);
+                break;
+            case SUCCESS:
+                getDialogHolder().dismissDialog();
+
+                // TODO: Should this logic be in the View?  If not how do we link the view models?
```

Comment:
I think it makes sense to have this in the View since it's acting as the glue binding all the different UI components together: we're moving from sign-in to credential input  which are two different "screens".